client/src/worker_registry/mod.rs

```diff
@@ -0,0 +1,288 @@
+//! This module enables the tracking of workers that are associated with a client instance.
+//! This is needed to implement Eager Workflow Start, a latency optimization in which the client,
+//!  after reserving a slot, directly forwards a WFT to a local worker.
+
+use rand::{seq::SliceRandom, thread_rng};
+use std::{collections::HashMap, fmt::Debug};
+use temporal_sdk_core_protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse;
+
+/// This trait wraps a worker, which provides WFT processing slots.
+#[cfg_attr(test, mockall::automock)]
+pub trait SlotProvider: Send + Sync + Debug {
```

Comment:
Typically you don't require supertraits that the trait itself doesn't actually need - rather, include the bounds in places where you need the trait to also satisfy those bounds.

Reply:
Got rid of Send+Sync, I had to keep Debug for SlotManager, which needs to implement  Debug for ConfiguredClient...

Reply:
Where does it require it? In any case, the same idea should apply, you only need to say `+ Debug` where needed in `ConfiguredClient`. But, it doesn't look like that's actually holding on to a `SlotProvider`?